tests/core/v5_1/alexandria/test_resource_queue.py

```diff
@@ -0,0 +1,77 @@
+import random
+
+import pytest
+import trio
+
+from ddht.v5_1.alexandria.resource_queue import ResourceQueue
+
+
+async def _yield(num: int = 10, base: int = 0):
+    for _ in range(random.randint(0, num) + base):
+        await trio.lowlevel.checkpoint()
+
+
+@pytest.mark.trio
+async def test_resource_queue():
+    known_resources = {"a", "b", "c", "d"}
+    queue = ResourceQueue(known_resources)
+
+    resources_in_use = set()
```

Comment:
I'm not sure what `resources_in_use` is meant to be testing. If I'm reading this right then there is no implementation of `ResourceQueue` which cause the `resources_in_use` assertions to fail.

Reply:
Yeah, it's a dirty way for me to test the "property" that any resource that is in use should never be allocated in a call to `queue.reserve`.  I recognize this is effectively a re-implementation of the business logic using stateful local variables.  I'm open to an alternate approach if you've got a suggestion.

Reply:
Ahh, sorry, I missed that there are 10 instances of worker, this makes sense now.

I bet some combination of [hypothesis](https://hypothesis.readthedocs.io/en/latest/stateful.html) or [sequencer](https://trio.readthedocs.io/en/v0.12.1/reference-testing.html#trio.testing.Sequencer) could help make this test simpler, but this already works so I definitely won't hold the review over it.